rumqttc/src/tls.rs

```diff
@@ -102,31 +107,31 @@ pub async fn rustls_connector(tls_config: &TlsConfiguration) -> Result<RustlsCon
 
             // Add der encoded client cert and key
             let mut config = if let Some(client) = client_auth.as_ref() {
+
                 let certs =
                     rustls_pemfile::certs(&mut BufReader::new(Cursor::new(client.0.clone())))?;
-                // load appropriate Key as per the user request. The underlying signature algorithm
+                if certs.is_empty() {
+                    return Err(Error::NoValidCertInChain);
```

Comment:
`NoValidCertInChain` was used to invalid / missing CA certs previously, so instead of creating new variant for `NoValidCACert`, we should create `NoValidClientCert` with `#[error("No valid certificate for client authentication in chain")]` .

If we want, we can just update the desc of `NoValidCertInChain` to say `#[error("No valid CA certificate provided")]`

wdyt?

PS: update changelog accordingly!

Reply:
Sounds good, then I will add `NoValidClientCertInChain`